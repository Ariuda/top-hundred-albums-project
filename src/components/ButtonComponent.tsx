import React from 'react';

interface Props {
    text?: string;
    onClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
    link?: string;
    className?: string;
}

export const ButtonComponent: React.FC<Props> = ({ text, onClick, link, className }) => {
    if (link) {
        return (
            <a href={link} className={`button ${className ? className : ''}`} target="_blank" rel="noreferrer">{text}</a>
        )
    }
    return <button className={`button ${className ? className : ''}`} onClick={onClick}>{text}</button>
    
};

export default ButtonComponent;