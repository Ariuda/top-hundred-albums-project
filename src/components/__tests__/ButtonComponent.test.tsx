import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { ButtonComponent } from '../ButtonComponent';

const props = {
    text: 'test',
    onClick: () => console.log('test'),
    link: 'test-link',
    className: 'test-class'
}

it('should render the button as a link when a link is passed', () => {
    const wrapper = shallow(<ButtonComponent text={props.text} link={props.link} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
});

it('should render the button as a link with an additional class when a link and a classname are passed', () => {
    const wrapper = shallow(<ButtonComponent text={props.text} link={props.link} className={props.className} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
});

it('should render the button as a button when no link is passed', () => {
    const wrapper = shallow(<ButtonComponent text={props.text} onClick={props.onClick} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
});

it('should render the button as a button with an additional class when there is no link and a classname are passed', () => {
    const wrapper = shallow(<ButtonComponent text={props.text} onClick={props.onClick} className={props.className} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
});