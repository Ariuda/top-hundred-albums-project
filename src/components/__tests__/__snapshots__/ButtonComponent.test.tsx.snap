// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the button as a button when no link is passed 1`] = `
<button
  className="button "
  onClick={[Function]}
>
  test
</button>
`;

exports[`should render the button as a button with an additional class when there is no link and a classname are passed 1`] = `
<button
  className="button test-class"
  onClick={[Function]}
>
  test
</button>
`;

exports[`should render the button as a link when a link is passed 1`] = `
<a
  className="button "
  href="test-link"
  rel="noreferrer"
  target="_blank"
>
  test
</a>
`;

exports[`should render the button as a link with an additional class when a link and a classname are passed 1`] = `
<a
  className="button test-class"
  href="test-link"
  rel="noreferrer"
  target="_blank"
>
  test
</a>
`;
